src/components/scene/ar-hit-test.js

```diff
@@ -1,5 +1,29 @@
+var register = require('../../core/component').registerComponent;
+var THREE = require('../../lib/three');
+var CAM_LAYER = 21;
+var previousFrameAnchors = new Set();
+var anchorToObject3D = new Map();
+var hitTestCache;
+var tempVec3 = new THREE.Vector3();
+var tempQuaternion = new THREE.Quaternion();
+var fakePose = {
+  transform: {
+    orientation: new THREE.Quaternion(),
+    position: new THREE.Vector3()
+  }
+};
+
+function applyPose (pose, object3D, offset) {
+  object3D.position.copy(pose.transform.position);
+  object3D.quaternion.copy(pose.transform.orientation);
+
+  tempVec3.copy(offset);
+  tempQuaternion.copy(pose.transform.orientation);
+  tempVec3.applyQuaternion(tempQuaternion);
+  object3D.position.sub(tempVec3);
+}
 
 function HitTest (renderer, options) {
```

Comment:
Maybe better to replace options with explicit parameters for clarity. Also adding a description on top of HitTest to document what the object does would be useful

Reply:
Sure, this is a hangover from when I had destructuring before.

Describing HitTest is a good idea since it is a wrapper around the WebXR HitTest API to handle the difference between WebXR Input Types